### PR TITLE
WIP: CNV-51570: Redirect from 'All projects' to accessible project for non-admin user

### PR DESCRIPTION
**Pull request by pcbailey**
## 📝 Description

Non-admin users shouldn't be shown 'All projects'. This PR redirects these users to an accessible project instead. This isn't a complete solution. This should happen automatically on the console side. I will create a PR on console as well.

Jira: https://issues.redhat.com/browse/CNV-51570

### CODE REVIEW

File: src/utils/hooks/useSetDefaultNonAdminUserProject/useSetDefaultNonAdminUserProject.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  const isAdmin = useIsAdmin();
+  const [activeNamespace, setActiveNamespace] = useActiveNamespace();
+  const [projects] = useProjects();
+
+  const nonDefaultUserProjects = projects.reduce((acc, project) => {
+    if (!isSystemNamespace(project)) acc.push(project);
+    return acc;
+  }, []);
+
+  if (isAdmin || projects.includes(activeNamespace)) return;
```
Comment (avivtur) [inline src/utils/hooks/useSetDefaultNonAdminUserProject/useSetDefaultNonAdminUserProject.ts:19]: Moving this before the reduce method will save a loop run

File: src/views/clusteroverview/ClusterOverviewPage.tsx [hunk 2]
```diff
@@ -17,6 +17,7 @@ import TopConsumersTab from './TopConsumersTab/TopConsumersTab';
 const ClusterOverviewPage: FC = () => {
   const { t } = useKubevirtTranslation();
   const isAdmin = useIsAdmin();
+  // useSetDefaultNonAdminUserProject();
```
Comment (avivtur) [inline src/views/clusteroverview/ClusterOverviewPage.tsx:20]: on purpose?

File: src/views/instancetypes/list/InstanceTypePage.tsx [hunk 3]
```diff
@@ -23,16 +25,21 @@ import '@kubevirt-utils/styles/list-managment-group.scss';
...
-  const activeNamespace = props?.namespace ?? ALL_NAMESPACES;
+  const [activeNamespace] = useActiveNamespace();
+  const currentNamespace = props?.namespace ?? activeNamespace ?? ALL_NAMESPACES;
 
   const isSearchPage = useIsSearchPage();
 
   const activeTabKey = useMemo(
-    () => (location?.pathname.includes(VirtualMachineClusterInstancetypeModel.kind) ? 0 : 1),
+    () =>
+      location?.pathname.includes(VirtualMachineClusterInstancetypeModel.kind) && isAdmin ? 0 : 1,
```
Comment (avivtur) [inline src/views/instancetypes/list/InstanceTypePage.tsx:38]: why do we need the isAdmin here too? won't this prevent from non-admin users to reach the VM create flow with IT?

**Review by avivtur — COMMENTED** [review]
This issue sounds more like a problem with console rather than our plugin issue, can explain to me the logic for us to remove manually the all-namespaces?

**Comment by pcbailey** [comment]
This bug has been moved to console: https://issues.redhat.com/browse/OCPBUGS-45944.

**Pull request closed without merging**